### PR TITLE
Basic Context Menu (behind Switch)

### PR DESCRIPTION
**Pull request by tylerjbainbridge**
Basic working context menu built with LexicalMenus so there's built in support for keyboard interaction, positioning, and an identical API to typeaheads, autoembeds, etc.

WIP demo

https://user-images.githubusercontent.com/13852400/230664047-34189629-2f43-47c4-9e62-0e436a06db7d.mov

### CODE REVIEW

**Comment by tylerjbainbridge** [comment]
> Hey, @tylerjbainbridge, any plans for updating this one, since the generic part is now merged?

Thanks for the reminder. I'll try to get this merged today.

**Review by ivailop7 — APPROVED** [review]
Wasn't sure, if I should review, since it's still a draft, but this looks pretty complete to me to what what's in the PR description

File: packages/lexical-react/src/LexicalContextMenuPlugin.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+import {LexicalMenu, MenuOption, useMenuAnchorRef} from './shared/LexicalMenu';
+
+export type ContextMenuRenderFn<TOption extends MenuOption> = (
+  anchorElementRef: MutableRefObject<HTMLElement | null>,
+  itemProps: {
+    selectedIndex: number | null;
+    selectOptionAndCleanUp: (option: TOption) => void;
+    setHighlightedIndex: (index: number) => void;
+    options: Array<TOption>;
```
Comment (zurfyx) [inline packages/lexical-react/src/LexicalContextMenuPlugin.tsx:29]: Are we planning to add the ability to have nested options?

File: packages/lexical-react/src/LexicalContextMenuPlugin.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      }
+    },
+    [onOpen, resolution],
+  );
+
+  const handleContextMenu = useCallback(
+    (event: MouseEvent) => {
+      event.preventDefault();
+      openNodeMenu({
+        getRect: () => new DOMRect(event.clientX, event.clientY, 1, 1),
```
Comment (zurfyx) [inline packages/lexical-react/src/LexicalContextMenuPlugin.tsx:89]: 1, 1?

**Review by zurfyx — APPROVED** [review]

File: packages/lexical-react/src/LexicalContextMenuPlugin.tsx (hunk 2 above)
Comment (tylerjbainbridge) [reply packages/lexical-react/src/LexicalContextMenuPlugin.tsx:89]: Kind of a hack, but the menus are positioned using a div (in this case 1px by 1px) that is replaced with React portal's content. I can add enums though.

**Pull request merged**